src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,97 @@
+package duke.task;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+import duke.util.ParsedData;
+
+/**
+ * Parent task class for all other Tasks.
+ */
+public abstract class Task implements Comparable<Task> {
+    protected final String description;
+    protected boolean completed;
+    protected Optional<LocalDateTime> dateTime;
+
+    protected Task(String description, Optional<LocalDateTime> dateTime) {
+        this.description = description;
+        this.completed = false;
+        this.dateTime = dateTime;
+    }
+
+    protected Task(String description) {
+        this(description, Optional.empty());
+    }
+
+    /**
+     * Marks the task as complete.
+     */
+    public void mark() {
+        completed = true;
+    }
+
+    /**
+     * Marks the task as incomplete.
+     */
+    public void unmark() {
+        completed = false;
+    }
+
+    /**
+     * Check if task is completed.
+     * 
+     * @return
+     */
+    public boolean isCompleted() {
+        return completed;
+    }
+
+    /**
+     * {@inheritDoc} Compares order based on data assigned to task.
+     * 
+     * @param o The other task
+     * @return a task is bigger if its time higher and biggest if non-existent
+     */
+    @Override
+    public int compareTo(Task o) {
```

Comment:
Perhaps a more descriptive name than "o" should be used?

Reply:
changed `o` -> `task`.
Although for under the scope of the constructor which is really short, I think this is ok 

>  Variables with a large scope should have long names, variables with a small scope can have short names.